cluster/azure/aks-api-server-access/kube_api_server_access.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+MODE=""
+IP=""
+RESOURCE_GROUP=""
+CLUSTER_NAME=""
+MODE_FLAGS=0
+USE_IP_LIST=0
+
+usage() {
+    echo "Usage: $?"
+    echo "  -a add IP address"
+    echo "  -r remove IP address"
+    echo "  -i <IP address>"
+    echo "  -g <Resource Group>"
+    echo "  -n <Cluster Name>"
+    echo "  -s <IP list>"
+    echo ""
+    echo "To add an IP address:"
+    echo "  $? -a -i <IP address> -g <Resource Group> -n <Cluster Name>"
+    echo ""
+    echo "To remove an IP address:"
+    echo "  $? -r -i <IP address> -g <Resource Group> -n <Cluster Name>"
+    echo ""
+    echo "Omitting the '-i' flag will use the current external IP address discovered using IP Chicken"
+    echo ""
+    echo "It is possible to replace '-i' with '-s' for adding IP addresses.  What '-s' does is replaces"
+    echo "all the values with the specified list.  This can also be used to set the list to null, thus"
+    echo "opening up the IP address range to all"
+    exit 1
+}
+
+# remove an ip address from a list
+subtract_ip()
+{
+    IP_TO_REMOVE=( $1 )
+    IP_LIST=( $2 )
+    OLDIFS="$IFS"
+    IFS=$'\n'
+    UPDATED_LIST=( $(grep -Fxv "${IP_TO_REMOVE[*]}" <<< "${IP_LIST[*]}") )
+    IFS="$OLDIFS"
+    echo "${UPDATED_LIST[*]}"
+}
+
+while getopts "ari:n:g:s:" OPTION; do
+    case $OPTION in
+    a)
+        MODE="add"
+        MODE_FLAGS=$((MODE_FLAGS+1))
+        ;;
+    r)
+        MODE="remove"
+        MODE_FLAGS=$((MODE_FLAGS+1))
+        ;;
+    i)
+        IP=$OPTARG
+        ;;
+    s)
+        IP_LIST=$OPTARG
+	USE_IP_LIST=1
+        ;;
+    n)
+        CLUSTER_NAME=$OPTARG
+        ;;
+    g)
+        RESOURCE_GROUP=$OPTARG
```

Comment:
I suppose all these are already guaranteed to be interpreted as a single argument, so no need to use `$OPTARG`? Nice!